arviz/tests/external_tests/test_data_pyro.py

```diff
@@ -123,3 +172,65 @@ def model_example_multiple_obs(y1=None, y2=None):
         fails = check_multiple_attrs(test_dict, inference_data)
         assert not fails
         assert not hasattr(inference_data.sample_stats, "log_likelihood")
+
+    def test_inference_data_constant_data(self):
+        import pyro.distributions as dist
+        from pyro.infer import MCMC, NUTS
+
+        x1 = 10
+        x2 = 12
+        y1 = torch.randn(10)
+
+        def model_constant_data(x, y1=None):
+            _x = pyro.sample("x", dist.Normal(1, 3))
+            pyro.sample("y1", dist.Normal(x * _x, 1), obs=y1)
+
+        nuts_kernel = NUTS(model_constant_data)
+        mcmc = MCMC(nuts_kernel, num_samples=10)
+        mcmc.run(x=x1, y1=y1)
+        posterior = mcmc.get_samples()
+        posterior_predictive = Predictive(model_constant_data, posterior)(x1)
+        predictions = Predictive(model_constant_data, posterior)(x2)
+        inference_data = from_pyro(
+            mcmc,
+            posterior_predictive=posterior_predictive,
+            predictions=predictions,
+            constant_data={"x1": x1},
+            predictions_constant_data={"x2": x2},
+        )
+        test_dict = {
+            "posterior": ["x"],
+            "posterior_predictive": ["y1"],
+            "sample_stats": ["diverging"],
+            "log_likelihood": ["y1"],
+            "predictions": ["y1"],
+            "observed_data": ["y1"],
+            "constant_data": ["x1"],
+            "predictions_constant_data": ["x2"],
+        }
+        fails = check_multiple_attrs(test_dict, inference_data)
+        assert not fails
+
+    def test_inference_data_dims(self, data, eight_schools_params, predictions_data):
+        inference_data = self.get_inference_data(data, eight_schools_params, predictions_data)
+        test_dict = {
+            "posterior": ["mu", "tau", "eta"],
+            "sample_stats": ["diverging"],
+            "posterior_predictive": ["obs"],
+            "predictions": ["obs"],
+            "prior": ["mu", "tau", "eta"],
+            "prior_predictive": ["obs"],
+        }
+        fails = check_multiple_attrs(test_dict, inference_data)
+        assert not fails
+
+        dims = inference_data.posterior_predictive["obs"].shape[2:]
+        pred_dims = inference_data.predictions["obs"].shape[2:]
+        assert dims == (8,)
+        assert pred_dims == (8,)
```

Comment:
`inference_data.posterior_predictive.dims["school"]` and `inference_data.predictions.dims["school_pred"]`, then the assert will be `dims == 8`

Reply:
This can be moved to the `test_inference_data` test above too, both start with the same code.